src/lib/manager/cloudFrontManager.ts

```diff
@@ -300,4 +320,32 @@ export class CloudFrontManager {
       policy: cr.AwsCustomResourcePolicy.fromSdkCalls({ resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE }),
     })
   }
+
+  /**
+   * @summary Method to provision a Cloudfront function
+   *
+   * @param {string} id scoped id of the resource
+   * @param {CommonConstruct} scope scope in which this resource is defined
+   * @param {CloudfrontFunctionProps} props
+   * @param {string} functionFilePath
+   */
+  public createCloudfrontFunction(
+    id: string,
+    scope: CommonConstruct,
+    props: CloudfrontFunctionProps,
+    functionFilePath: string
+  ) {
+    const cloudfrontFunction = new cloudfront.Function(scope, `${id}`, {
+      code: cloudfront.FunctionCode.fromFile({
+        filePath: functionFilePath,
+      }),
+      comment: props.comment,
+      functionName: `${id}-${props.functionName}-${scope.props.stage}`,
```

Comment:
shouldn't use `ID` in the function name but rather just use the `functionName` with `stage`. 
... different projects will have different conventions and the functionNames should be free to choose from